x86_symbolic/src/Data/Macaw/X86/Symbolic/ABI/SysV.hs

```diff
@@ -116,11 +170,50 @@ writeSpilledArgs bak mem sp spilledArgs = do
   eight <- WI.bvLit sym ptrRepr (BVS.mkBV WI.knownNat 8)
   let i64 = MM.bitvectorType (Bytes.toBytes (64 :: Int))
   let ?ptrWidth = ptrRepr
+
+  -- First, align to 8 bytes. In all probability, this is a no-op.
+  let align8 = CLD.exponentToAlignment 3  -- 2^3 = 8
+  StackPointer spAligned8 <- alignStackPointer sym align8 sp
+
+  -- At this point, exactly one of `spAligned8` or `spAligned8 + 8` is 16-byte
+  -- aligned. We need to ensure that, after writing the argument list, the stack
+  -- pointer will be 16-byte aligned.
+  --
+  -- Depending on the current alignment and partity of the number of spilled
+  -- arguments, we need to:
+  --
+  -- - further align `rsp` to 16 bytes (call this choice a16)
+  -- - add 8 so that `rsp + 8` is aligned to 16 bytes (call this +8)
```

Comment:
Given that the stack grows downwards, I'm somewhat surprised that you'd _add_ 8 to align the stack pointer to 16 bytes. Shouldn't you _subtract_ 8 instead? (I might be missing something here.)

Reply:
Whoops, yep! Added a new function for allocating stack space that makes this more explicit.